packages/http-client/src/http-client-types.ts

```diff
@@ -10,6 +12,7 @@ export interface DestinationHttpRequestConfig {
 
 /**
  * HTTP Methods supported by the http-client.
+ * @internal
```

Comment:
I would keep this as part of `HttpRequestConfig`. Also, there is a [request](https://github.com/SAP/cloud-sdk-js/issues/1323) for supporting overwriting methods.